### PR TITLE
Error and warning dialog refinements

### PR DESCRIPTION
**Pull request by niik**
This implements the dialog refinements outlined in https://github.com/desktop/desktop/pull/2185#issuecomment-314513347. Note that this only affects warning and error type dialogs.

Dialog changes

 - Moved the icon from the header down to the content
 - Made custom 24px versions of the stop and alert octicons
 - Removed colored top border
 - Left align dialog title
 - Moved progress indicator to the right hand side of the title
 - Removed custom text color of title in error and warning dialogs
 
# Before

image

image

# After

image

image

image

### CODE REVIEW

**Comment by niik** [comment]
> @niik Does this need @donokuda review or just code review?

Yeah, I'd love a quick @donokuda look-over. Who wouldn't

File: app/styles/ui/_dialog.scss [hunk 1]
```diff
@@ -160,6 +155,52 @@ dialog {
...
+      // add another double spacing plus the width of the icon to ensure
+      // we have a decently horizontally centered placement of the icon.
+      padding-left: calc(var(--spacing-double) + var(--dialog-icon-size));
+
+      // Ensure that the dialog contents always have room for the icon,
+      // account for two double spacers at top and bottom plus the 5px
+      // icon offset (margin-top) and the size of the icon itself.
+      min-height: calc(var(--spacing-double) * 2 + var(--spacing-half) + var(--dialog-icon-size));
+
+      // We're creating an opaque 24 by 24px div with the background color
```
Comment (joshaber) [inline app/styles/ui/_dialog.scss]: hqdefault

**Comment by niik** [comment]
@joshaber I suspect @donokuda might be a bit swamped so I say we merge this as-is and I'll go through it with him in the design review next week and we'll make any tweaks in a follow-up PR. We've already agreed on the general direction.

**Comment by joshaber** [comment]
👍 💯

**Pull request merged**